pkg/resources/authentication_policy.go

```diff
@@ -246,127 +274,124 @@ func ReadContextAuthenticationPolicy(ctx context.Context, d *schema.ResourceData
-			baseReq.WithUnset(*unsetReq)
-		} else {
-			setReq := sdk.NewAuthenticationPolicySetRequest().WithMfaAuthenticationMethods(mfaAuthenticationMethodsValues)
-			baseReq.WithSet(*setReq)
-		}
+	// change to mfa authentication methods
+	if d.HasChange("mfa_authentication_methods") {
+		if v, ok := d.GetOk("mfa_authentication_methods"); ok {
+			mfaAuthenticationMethods := expandStringList(v.(*schema.Set).List())
+			mfaAuthenticationMethodsValues := make([]sdk.MfaAuthenticationMethods, len(mfaAuthenticationMethods))
+			for i, v := range mfaAuthenticationMethods {
+				mfaAuthenticationMethodsValues[i] = sdk.MfaAuthenticationMethods{Method: sdk.MfaAuthenticationMethodsOption(v)}
+			}
 
-		if err := client.AuthenticationPolicies.Alter(ctx, baseReq); err != nil {
-			return diag.FromErr(err)
+			if len(mfaAuthenticationMethodsValues) == 0 {
+				unset.MfaAuthenticationMethods = sdk.Bool(true)
+			} else {
+				set.MfaAuthenticationMethods = mfaAuthenticationMethodsValues
+			}
 		}
 	}
 
 	// change to mfa enrollment
 	if d.HasChange("mfa_enrollment") {
-		mfaEnrollment := d.Get("mfa_enrollment").(string)
-
-		baseReq := sdk.NewAlterAuthenticationPolicyRequest(id)
-		if len(mfaEnrollment) == 0 {
-			unsetReq := sdk.NewAuthenticationPolicyUnsetRequest().WithMfaEnrollment(*sdk.Bool(true))
-			baseReq.WithUnset(*unsetReq)
+		if v, ok := d.GetOk("mfa_enrollment"); ok {
+			set.MfaEnrollment = sdk.Pointer(sdk.MfaEnrollmentOption(v.(string)))
 		} else {
-			setReq := sdk.NewAuthenticationPolicySetRequest().WithMfaEnrollment(sdk.MfaEnrollmentOption(mfaEnrollment))
-			baseReq.WithSet(*setReq)
-		}
-
-		if err := client.AuthenticationPolicies.Alter(ctx, baseReq); err != nil {
-			return diag.FromErr(err)
+			unset.MfaEnrollment = sdk.Bool(true)
 		}
 	}
 
 	// change to client types
 	if d.HasChange("client_types") {
-		clientTypes := expandStringList(d.Get("client_types").(*schema.Set).List())
-		clientTypesValues := make([]sdk.ClientTypes, len(clientTypes))
-		for i, v := range clientTypes {
-			clientTypesValues[i] = sdk.ClientTypes{ClientType: sdk.ClientTypesOption(v)}
-		}
-
-		baseReq := sdk.NewAlterAuthenticationPolicyRequest(id)
-		if len(clientTypesValues) == 0 {
-			unsetReq := sdk.NewAuthenticationPolicyUnsetRequest().WithClientTypes(*sdk.Bool(true))
-			baseReq.WithUnset(*unsetReq)
-		} else {
-			setReq := sdk.NewAuthenticationPolicySetRequest().WithClientTypes(clientTypesValues)
-			baseReq.WithSet(*setReq)
-		}
+		if v, ok := d.GetOk("client_types"); ok {
+			clientTypes := expandStringList(v.(*schema.Set).List())
+			clientTypesValues := make([]sdk.ClientTypes, len(clientTypes))
+			for i, v := range clientTypes {
+				clientTypesValues[i] = sdk.ClientTypes{ClientType: sdk.ClientTypesOption(v)}
+			}
 
-		if err := client.AuthenticationPolicies.Alter(ctx, baseReq); err != nil {
-			return diag.FromErr(err)
+			if len(clientTypesValues) == 0 {
+				unset.ClientTypes = sdk.Bool(true)
```

Comment:
You can do `unset.WithClientTypes(true)`. The same applies to `set` requests, and for all the fields.

Reply:
Done